eo-phi-normalizer/src/Language/EO/Phi/Rules/Common.hs

```diff
@@ -127,3 +129,34 @@ lookupBinding a (AlphaBinding a' object : bindings)
   | a == a' = Just object
   | otherwise = lookupBinding a bindings
 lookupBinding _ _ = Nothing
+
+objectBindings :: Object -> [Binding]
+objectBindings (Formation bs) = bs
+objectBindings _ = []
+
+nuCount :: Object -> Int
```

Comment:
This function apparently does not account for vertex indices in arguments of applications.